Enhance Dashboard UI with KPI Cards, Visualizations, Theme Toggle & Improved Run Mapping

dakshita-singh opened the pull request:
## Overview

This pull request introduces a major enhancement to the TestCraft Dashboard, focusing on improving usability, readability, and visual clarity for both technical and non-technical users. The update includes a redesigned dashboard layout, new KPI cards, multiple visual charts, theme support, and a simplified test run representation.

### Key Improvements
1. Modernized Dashboard UI
 - Redesigned overall layout with clean spacing, improved hierarchy, and responsive grid.
 - Fixed duplicate header issue and standardized dashboard title.

2. Light/Dark Theme Toggle
 - Added global theme toggle in the dashboard.
 - All components now adapt consistently to theme changes.
 - Improved contrast, readability, and color handling across charts and tables.

3. KPI Summary Cards
 - Added four new KPI cards:
 - Total Tests Executed
 - Passing Tests
 - Failed Tests
 - Average Duration
 - Each card dynamically updates based on run data.

4. Improved Recent Test Runs Table
- Replaced confusing test IDs (e.g., AC-LOGIN-001) with friendly test names.
- Added click-through link for feature/application context.
- Clean, simple layout for non-technical stakeholders.

5. New Visualization Components
- The dashboard now includes multiple charts inside reusable <ChartCard> containers:
- Run Status Distribution
- Daily Run Trend
- Average Run Duration
- Environment Distribution

6. Added Supporting Modules
- Test Pipeline Funnel
- Flaky Tests Heatmap
- Visual Regression Viewer placeholder

7. Improved Sample Run Mapping
- Added mapping layer for converting raw test IDs → human-readable names.
- Ensures the dashboard is easily understood by non-technical users.
- Why This Update Matters
 - This update significantly enhances the clarity and professionalism of the TestCraft Dashboard.

It helps:
- QA Managers
- Stakeholders
- Non-technical team members

to quickly understand release quality, test activity, and pipeline health without decoding technical IDs.

<img width="959" height="437" alt="image" src="https://github.com/user-attachments/assets/fffc71c3-8b40-46c8-81be-c81c9620aa1f" />
<img width="959" height="450" alt="image" src="https://github.com/user-attachments/assets/f3ca8088-ee15-4526-a9ac-ff553ca83905" />
<img width="838" height="227" alt="image" src="https://github.com/user-attachments/assets/3d4bb245-0420-4971-b392-a443ebf4170e" />
<img width="776" height="385" alt="image" src="https://github.com/user-attachments/assets/8c8fd2b1-6c87-4324-a21c-d8922060f52d" />
<img width="907" height="349" alt="image" src="https://github.com/user-attachments/assets/d4c0d041-3766-4e4c-bf4a-0d942afdf8e3" />

dakshita-singh committed:
Add Dashboard UI Proposal for TestCraft

This document outlines the design proposal for the Dashboard UI of TestCraft, detailing its purpose, design principles, layout, unique features, components, and data model.

Vaibosnikov committed:
feat(dashboard-ui): add UI scaffold (React + Vite + Tailwind) with shell, components and mock runs

Vaibosnikov commented:
@dakshita-singh Please take care of the merge conflicts and markdown formatting properly. I'll try to make the branch mergeable again by resolving your conflicts.

Thanks!

Vaibosnikov committed:
Merge branch 'dev' into feature/dashboard-ui

Vaibosnikov reviewed (approved):
Thanks @dakshita-singh for the refinement. Everything looks great in picture. Make sure each charts functionality and what meaning it conveys, this should be aware each of us.

Anshra-dev reviewed (approved):
All looks good

Vaibosnikov committed:
Refactor package.json and component files for dashboard UI

- Updated package.json to rename project from "testcraft-dashboard" to "testcraft-dashboard-ui" and removed conflicting sections.
- Modified Header component to reflect the new project name and updated description.
- Refactored RunsTable component to simplify the no runs message and removed unused code.
- Cleaned up sampleRuns.json by removing unnecessary conflict markers.
- Revised apiClient.js to streamline the fetchRuns function and removed outdated comments.
- Adjusted vite.config.js to include HMR overlay configuration.
